app/entitycore_helper.py

```diff
@@ -0,0 +1,334 @@
+    morph_dir = pathlib.Path(morph_dir)
+    morph_dir.mkdir(parents=True, exist_ok=True)
+    asset_id = None
+    asset_path = None
+    for asset in morphology.assets:
+        if file_type is None or file_type in asset.content_type:
+            asset_id = asset.id
+            asset_path = asset.path
+            break
+    if asset_id is None:
+        ftype = file_type or ""
+        raise ValueError(f"No {ftype} file found in the morphology {morphology.name}.")
+    morph_out_path = morph_dir / asset_path
+    client.download_file(
+        asset_id=asset_id,
+        entity_id=morphology.id,
+        entity_type=ReconstructionMorphology,
+        token=access_token,
+        output_path=morph_out_path,
+    )
+    return morph_out_path
+
+
+def get_holding_and_threshold(calibration_result):
+    """Get holding and threshold currents from the MEModel.
+
+    Args:
+        calibration_result (MEModelCalibrationResult or None): Calibration result entitysdk object
+    """
+    holding_current = None
+    threshold_current = None
+    if calibration_result is not None:
+        holding_current = calibration_result.holding_current
+        threshold_current = calibration_result.threshold_current
+
+    return holding_current, threshold_current
+
+
+def download_memodel(client, access_token, memodel_id=None, memodel_name=None):
+    """Download MEModel
+
+    Args:
+        client (Client): EntitySDK client
+        access_token (str): access token for authentication
+        memodel_id (str or None): id of the MEModel to download
+        memodel_name (str or None): name of the MEModel to download
+    """
+    if memodel_id is None and memodel_name is None:
+        raise ValueError("Either memodel_id or memodel_name must be provided.")
+    if memodel_id is not None:
+        memodel = client.get_entity(
+            entity_type=MEModel,
+            entity_id=memodel_id,
+            token=access_token,
+        )
+    else:
+        iterator = client.search_entity(
+            entity_type=MEModel,
+            query={"name": memodel_name},
+            token=access_token,
+            limit=1,
+        )
+        memodel = next(iterator)
+
+    morphology = memodel.morphology
+    # we have to get the emodel to get the ion channel models.
+    emodel = client.get_entity(entity_id=memodel.emodel.id, entity_type=EModel, token=access_token)
+
+    holding_current, threshold_current = get_holding_and_threshold(memodel.calibration_result)
+
+    # + 2 for hoc and morphology
+    # len of ion_channel_models should be around 10 for most cases,
+    # and always < 100, even for genetic models
+    max_workers = len(emodel.ion_channel_models) + 2
+    with ThreadPoolExecutor(max_workers=max_workers) as executor:
+        hoc_future = executor.submit(download_hoc, emodel, client, access_token, "./hoc")
+        morph_future = executor.submit(
+            download_morphology, morphology, client, access_token, "./morphology"
```

Comment:
I see that the default morphology format set in the function itself is `asc` and it's not provided in this call, do we expect an asc morphology to be present in all models in the platform?

In the nexus helper there is a logic to select the format used in model optimisation as for some of the converted `swc` morphologies simulations were failing, just wondering if we need to do the same here.

Reply:
You are right, I should change this, we cannot rely on expecting the .asc to always be present.
Where is the format selection in the nexus handler? I could only find a hardcoded `morph_format = "swc"`

Reply:
alright, I implemented the fallback in the latest commits